kex-core/src/main/kotlin/org/vorpal/research/kex/parameters/Parameters.kt

```diff
@@ -92,3 +95,64 @@ fun Parameters<Descriptor>.filterIgnoredStatic(): Parameters<Descriptor> {
         }
     return Parameters(instance, arguments, filteredStatics)
 }
+
+fun Parameters<Descriptor>.replaceNullsWithDefaultValues(method: Method, cm: ClassManager): Parameters<Descriptor> {
+    val parametersWithoutNulls = arguments
+        .zip(method.argTypes)
+        .map { (descriptor, type) -> transformExistingDescriptor(descriptor, type.kexType, cm) }
+    return Parameters(instance, parametersWithoutNulls, statics)
+}
+
+private fun transformExistingDescriptor(descriptor: Descriptor, type: KexType, cm: ClassManager, alreadyCopied: MutableMap<Descriptor, Descriptor> = mutableMapOf()): Descriptor {
+    if (descriptor in alreadyCopied) return alreadyCopied[descriptor]!!
+    return when (descriptor) {
+        is ConstantDescriptor.Null -> type.newInstance(cm)
+        is ConstantDescriptor -> descriptor
+        is ArrayDescriptor -> ArrayDescriptor(descriptor.elementType, descriptor.length)
+            .apply {
+                (0 until descriptor.length).forEach { index ->
+                    this[index] = transformExistingDescriptor(
+                        descriptor[index] ?: ConstantDescriptor.Null,
+                        descriptor.elementType,
+                        cm,
+                        alreadyCopied
+                    )
+                }
+            }
+
+        is ObjectDescriptor -> ObjectDescriptor(descriptor.klass).apply {
+            descriptor.fields.forEach { (key, desc) ->
+                val (name, fieldType) = key
+                set(name to fieldType, transformExistingDescriptor(desc, fieldType, cm, alreadyCopied))
+            }
+        }
+
+        is ClassDescriptor -> ClassDescriptor(descriptor.klass).apply {
+            descriptor.fields.forEach { (key, desc) ->
+                val (name, fieldType) = key
+                set(name to fieldType, transformExistingDescriptor(desc, fieldType, cm, alreadyCopied))
+            }
+        }
+
+        is MockDescriptor -> MockDescriptor(descriptor.klass).apply {
+            descriptor.fields.forEach { (key, desc) ->
+                val (name, fieldType) = key
+                set(name to fieldType, transformExistingDescriptor(desc, fieldType, cm, alreadyCopied))
+            }
+        }
+    }.also { alreadyCopied[descriptor] = it }
```

Comment:
You need to add `descriptor` to `alreadyCopied` before calling `transformExistingDescriptor` recursively.
Because currently your implementaion will not protect you from loops inside object, e.g.:
```
x = foo.Node {
   (link, foo.Node) = x
}
```